src/zenml/integrations/skypilot/orchestrators/skypilot_base_vm_orchestrator.py

```diff
@@ -271,29 +272,33 @@ def prepare_or_run_pipeline(
                 f"sudo docker login --username $DOCKER_USERNAME --password "
                 f"$DOCKER_PASSWORD {stack.container_registry.config.uri}"
             )
-            task_envs = {
-                "DOCKER_USERNAME": docker_username,
-                "DOCKER_PASSWORD": docker_password,
-            }
+            task_envs["DOCKER_USERNAME"] = docker_username
+            task_envs["DOCKER_PASSWORD"] = docker_password
         else:
             setup = None
-            task_envs = None
 
         # Run the entire pipeline
 
         # Set the service connector AWS profile ENV variable
         self.prepare_environment_variable(set=True)
 
         try:
+            if isinstance(self.cloud, sky.clouds.Kubernetes):
+                run_command = f"${{VIRTUAL_ENV:+$VIRTUAL_ENV/bin/}}{entrypoint_str} {arguments_str}"
```

Comment:
Which shell should this run in? Doesn't work for me somehow

Reply:
Ah it's the double brackets because of the f-string, my bad. If you tested this and it works, feel free to merge.